libfwupdplugin/fu-smbios.c

```diff
@@ -89,6 +93,76 @@ fu_smbios_convert_dt_string (FuSmbios *self, guint8 type, guint8 offset,
 	fu_smbios_convert_dt_value (self, type, offset, item->strings->len);
 }
 
+#if defined (__FreeBSD__) || defined (__DragonFly__)
+static void
+fu_smbios_kenv_sysctl_string (FuSmbios *self, guint8 type, guint8 offset,
+				const gchar *buf, gsize bufsz)
+{
+	FuSmbiosItem *item = g_ptr_array_index (self->items, type);
+
+	/* add to strtab */
+	g_ptr_array_add (item->strings, g_strndup (buf, bufsz));
+	fu_smbios_convert_dt_value (self, type, offset, item->strings->len);
+}
+
+static gboolean
+fu_smbios_kenv_lookup (const char *sminfo, gchar *buf, gsize bufsz)
+{
+	if (kenv (KENV_GET, sminfo, buf, bufsz - 1) == -1) {
+		g_debug ("cannot get DMI request for %s", sminfo);
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static void
+fu_smbios_convert_kenv_string (FuSmbios *self, guint8 type, guint8 offset,
+				const char *sminfo)
+{
+	/* Maximum value length - 128 */
+	gchar buf[128] = { '\0' };
+	gsize bufsz = sizeof(buf);
+
+	if (fu_smbios_kenv_lookup (sminfo, buf, bufsz))
```

Comment:
since this can fail, I think that `fu_smbios_convert_kenv_string` should probably be boolean, and you should be passing/setting error along all these functions.

Reply:
Done.